### PR TITLE
rpc: Improve error message for getblock invalid datatype.

### LINKED ISSUE #21717: Invalid error message for getblock
**Expected behavior**

When I make a json-rpc call to `getblock` with:

```json
{
    "jsonrpc": "1.0",
    "id": "curltest",
    "method": "getblock",
    "params": ["37351c7060ec3a149af1bbc9ebcc436c0ee652f96559ef1fe9bc45d11000f061", "2"]
}
```
An error should be returned with the [expected data type](https://developer.bitcoin.org/reference/rpc/getblock.html#argument-2-verbosity)(number) for `verbosity`.

**Actual behavior**

Received `"message": "JSON value is not a boolean as expected"` as an error message which is not the [expected data type](number) (https://developer.bitcoin.org/reference/rpc/getblock.html#argument-2-verbosity) for `verbosity`

**To reproduce**

Use this curl on a node

```
curl --location --request POST "127.0.0.1:8332" \
--header "Content-Type: application/json" \
--header "Authorization: Basic YOUR_AUTH" \
--data "{
    \"jsonrpc\": \"1.0\",
    \"id\": \"curltest\",
    \"method\": \"getblock\",
    \"params\": [\"37351c7060ec3a149af1bbc9ebcc436c0ee652f96559ef1fe9bc45d11000f061\", \"2\"]
}"
```

**System information**
* self-compiled

### PR DESCRIPTION
**Pull request by klementtan**
Improve error messages for getblock invalid datatype. 

fixes: #21717

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -938,8 +938,10 @@ static RPCHelpMan getblock()
     if (!request.params[1].isNull()) {
         if(request.params[1].isNum())
```
Comment (MarcoFalke) [inline src/rpc/blockchain.cpp:939]: might be easier to swap the if branches and make this branch bool?

Comment (klementtan) [reply src/rpc/blockchain.cpp]: Yup you're right. Updated the PR according to your comments.

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -936,10 +936,10 @@ static RPCHelpMan getblock()
 
     int verbosity = 1;
     if (!request.params[1].isNull()) {
-        if(request.params[1].isNum())
-            verbosity = request.params[1].get_int();
-        else
+        if(request.params[1].isBool())
```
Comment (MarcoFalke) [inline src/rpc/blockchain.cpp:939]:
```suggestion
        if (request.params[1].isBool()) {
```

please use the new style per the dev notes

**Review by maflcko — APPROVED** [review]
Looks good. A test would be nice, if possible.

File: src/rpc/blockchain.cpp (hunk 2 above)
Comment (klementtan) [reply src/rpc/blockchain.cpp]: Added tests and updated the code to the new style

File: test/functional/rpc_blockchain.py [hunk 3]
```diff
@@ -410,6 +411,13 @@ def _test_getblock(self):
         self.log.info("Test that getblock with verbosity 2 still works with pruned Undo data")
         datadir = get_datadir_path(self.options.tmpdir, 0)
 
+        try:
+            self.log.info("Test that getblock with invlaid verbosity type returns proper error message")
+            node.getblock(blockhash, "2")
```
Comment (MarcoFalke) [inline test/functional/rpc_blockchain.py:416]: You can use the `assert_raises_rpc_error` helper

**Review by maflcko — APPROVED** [review]

File: test/functional/rpc_blockchain.py (hunk 3 above)
Comment (klementtan) [reply test/functional/rpc_blockchain.py]: Okie updated to use `assert_raises_rpc_error` instead.

**Comment by MarcoFalke** [comment]
review ACK 89cbab16139d91c4bcf2a7308b8b21afb0efe00e

File: test/functional/rpc_blockchain.py [hunk 4]
```diff
@@ -410,6 +410,9 @@ def _test_getblock(self):
         self.log.info("Test that getblock with verbosity 2 still works with pruned Undo data")
         datadir = get_datadir_path(self.options.tmpdir, 0)
 
+        self.log.info("Test that getblock with invlaid verbosity type returns proper error message")
```
Comment (theStack) [inline test/functional/rpc_blockchain.py:413]:
```suggestion
        self.log.info("Test that getblock with invalid verbosity type returns proper error message")
```

**Review by theStack — COMMENTED** [review]
Thanks for fixing, welcome as a new contributor! 🎉 

Almost-ACK, there are two typos: one in the commit subject (s/getbloack/getblock/), and one in the test log message (s/invlaid/invalid/).

File: test/functional/rpc_blockchain.py (hunk 4 above)
Comment (klementtan) [reply test/functional/rpc_blockchain.py]: My apologies for that. Have updated the PR to fix the typos. Thanks!

**Review by theStack — APPROVED** [review]
ACK a41149426168b8ea96099f10576022c6a09033d1

**Review by promag — COMMENTED** [review]
Code review ACK a41149426168b8ea96099f10576022c6a09033d1.

**Comment by instagibbs** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/21718/commits/a41149426168b8ea96099f10576022c6a09033d1

**Pull request merged**